### ISSUE TITLE
a11y: Checkbox labels not read in Firefox + NVDA

### ISSUE THREAD

**Issue by silvalaura** [cengage/react-magma#1122]
**Describe the bug**
The screen reader does not read the checkbox label. This is only reproduced for this combination of Firefox 115/116/117 + NVDA + Windows 10/11. All other browsers and screen readers work fine.

**To Reproduce**

**Expected behavior**
Screenreader is reads checkbox label

**Screenshots**
![firefox_sandbox](https://github.com/cengage/react-magma/assets/91160746/429c06f3-bf52-4445-80fc-b12f4d755c8b)
![chrome_reproducing_issue](https://github.com/cengage/react-magma/assets/91160746/8f4a7a5b-b766-4eff-8625-61b3a8554b3a)
![firefox_reproducing_issue](https://github.com/cengage/react-magma/assets/91160746/b1849dd1-9dc7-4d40-8964-5247b0bac71f)

**Comment by silvalaura** [comment]
I don't have that version of Firefox, but confirmed that labels are read on the latest FF with NVDA.
